x-pack/plugins/apm/public/components/app/service_overview/latency_chart/index.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useCallback, useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import { useUrlParams } from '../../../../context/url_params_context/use_url_params';
+import { FETCH_STATUS } from '../../../../hooks/use_fetcher';
+import { useTransactionChartsFetcher } from '../../../../hooks/use_transaction_charts_fetcher';
+import { TimeseriesChart } from '../../../shared/charts/timeseries_chart';
+import { getResponseTimeTickFormatter } from '../../../shared/charts/transaction_charts/helper';
+import { useFormatter } from '../../../shared/charts/transaction_charts/use_formatter';
+import { fromQuery, toQuery } from '../../../shared/Links/url_helpers';
+
+interface Props {
+  height?: number;
+}
+
+export type AggregationType = 'avg' | 'p95' | 'p99';
+
+const options: Array<{ value: AggregationType; text: string }> = [
+  { value: 'avg', text: 'Average' },
+  { value: 'p95', text: '95th percentile' },
+  { value: 'p99', text: '99th percentile' },
+];
+
+export function LatencyChart({ height }: Props) {
+  const history = useHistory();
+  const { urlParams } = useUrlParams();
+  const { aggregationType } = urlParams;
+
+  const {
+    transactionChartsData,
+    transactionChartsStatus,
+  } = useTransactionChartsFetcher();
+  // const { responseTimeSeries, anomalySeries } = transactionChartsData;
+
+  // const timeSeries = responseTimeSeries
+  //   ? [responseTimeSeries[selectedOption]]
+  //   : [];
+  // const { formatter } = useFormatter(timeSeries);
+  // const options = responseTimeSeries?.map(({ title }, index) => ({
+  //   value: index,
+  //   text: title,
+  // }));
+
+  const updateAggregationType = useCallback(
+    (selectedAggregationType: string) => {
+      const newLocation = {
+        ...history.location,
+        search: fromQuery({
+          ...toQuery(history.location.search),
+          aggregationType: selectedAggregationType,
+        }),
+      };
+      history.push(newLocation);
+    },
+    [history]
+  );
+
+  return (
+    <EuiFlexGroup direction="column">
+      <EuiFlexItem>
+        <EuiFlexGroup alignItems="center">
+          <EuiFlexItem grow={false}>
+            <EuiTitle size="xs">
+              <h2>
+                {i18n.translate('xpack.apm.serviceOverview.latencyChartTitle', {
+                  defaultMessage: 'Latency',
+                })}
+              </h2>
+            </EuiTitle>
+          </EuiFlexItem>
+          <EuiFlexItem grow={false}>
+            <EuiSelect
+              compressed
+              // isLoading={transactionChartsStatus === FETCH_STATUS.LOADING}
+              prepend={i18n.translate(
+                'xpack.apm.serviceOverview.latencyChartTitle.prepend',
+                { defaultMessage: 'Metric' }
+              )}
+              options={options}
+              value={aggregationType}
+              onChange={(nextOption) => {
+                updateAggregationType(nextOption.target.value);
+              }}
```

Comment:
Any reason to extract this to a separate function? Also, you can use `push` from url_helpers:



```suggestion
              onChange={(nextOption) => {
                urlHelpers.push({ aggregationType: nextOption.target.value })
              }}
```

Reply:
thanks, I didn't know about `urlHelpers.push`.

Reply:
I also only added it recently :)